build/patches/Flag-for-always-desktop-mode.patch

```diff
@@ -1,52 +1,201 @@
+ .../strings/android_chrome_strings.grd        |  35 ++++
+ chrome/common/pref_names.cc                   |  13 ++
+ chrome/common/pref_names.h                    |   8 +
+ .../android/browser_startup_controller.cc     |  35 ++++
+ .../navigation_controller_android.cc          |   6 +-
+ .../navigation_controller_android.h           |   3 +-
+ .../renderer_host/render_process_host_impl.cc |   1 +
+ .../browser/web_contents/web_contents_impl.cc |   4 +-
+ .../framehost/NavigationControllerImpl.java   |   6 +-
+ 28 files changed, 748 insertions(+), 30 deletions(-)
+ create mode 100644 chrome/android/java/res/layout/custom_useragent_preferences.xml
  create mode 100644 chrome/android/java/res/xml/useragent_preferences.xml
  create mode 100644 chrome/android/java/src/org/chromium/chrome/browser/settings/UserAgentPreferences.java
 
+diff --git a/base/base_switches.cc b/base/base_switches.cc
+--- a/base/base_switches.cc
++++ b/base/base_switches.cc
+@@ -160,4 +160,6 @@ const char kForceFieldTrialParams[] = "force-fieldtrial-params";
+ const char kEnableThreadInstructionCount[] = "enable-thread-instruction-count";
+ #endif
+ 
++const char kDesktopModeViewportMetaEnabled[] = "dm-viewport-meta-enabled";
++
+ }  // namespace switches
+diff --git a/base/base_switches.h b/base/base_switches.h
+--- a/base/base_switches.h
++++ b/base/base_switches.h
+@@ -59,6 +59,8 @@ extern const char kForceFieldTrialParams[];
+ extern const char kEnableThreadInstructionCount[];
+ #endif
+ 
++extern const char kDesktopModeViewportMetaEnabled[];
++
+ }  // namespace switches
+ 
+ #endif  // BASE_BASE_SWITCHES_H_
 diff --git a/chrome/android/chrome_java_resources.gni b/chrome/android/chrome_java_resources.gni
 --- a/chrome/android/chrome_java_resources.gni
 +++ b/chrome/android/chrome_java_resources.gni
-@@ -1068,4 +1068,5 @@ chrome_java_resources = [
+@@ -1068,4 +1068,6 @@ chrome_java_resources = [
    "java/res/xml/sync_and_services_preferences.xml",
    "java/res/xml/theme_preferences.xml",
    "java/res/xml/tracing_preferences.xml",
 +  "java/res/xml/useragent_preferences.xml",
++  "java/res/layout/custom_useragent_preferences.xml",
  ]
 diff --git a/chrome/android/chrome_java_sources.gni b/chrome/android/chrome_java_sources.gni
 --- a/chrome/android/chrome_java_sources.gni
 +++ b/chrome/android/chrome_java_sources.gni
-@@ -1322,6 +1322,7 @@ chrome_java_sources = [
-   "java/src/org/chromium/chrome/browser/privacy/settings/PrivacyPreferencesManager.java",
-   "java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java",
-   "java/src/org/chromium/chrome/browser/provider/BaseColumns.java",
+@@ -1308,6 +1308,7 @@ chrome_java_sources = [
+   "java/src/org/chromium/chrome/browser/photo_picker/DecoderService.java",
+   "java/src/org/chromium/chrome/browser/settings/AdBlockEditor.java",
+   "java/src/org/chromium/chrome/browser/settings/AdBlockPreferences.java",
 +  "java/src/org/chromium/chrome/browser/settings/UserAgentPreferences.java",
```

Comment:
Can you move this elsewhere? Being so close to files changed in another patch will cause problems.